tests/integration/telemetry/stats/prometheus/nullvm/accesslogs_test.go

```diff
@@ -51,15 +52,55 @@ func TestAccessLogs(t *testing.T) {
 		Features("observability.telemetry.logging").
 		Run(func(t framework.TestContext) {
 			t.NewSubTest("enabled").Run(func(t framework.TestContext) {
+				applyTelemetryResource(t, true)
 				runAccessLogsTests(t, true)
+				deleteTelemetryResource(t, true)
 			})
 			t.NewSubTest("disabled").Run(func(t framework.TestContext) {
+				applyTelemetryResource(t, false)
 				runAccessLogsTests(t, false)
+				deleteTelemetryResource(t, false)
 			})
 		})
 }
 
-func runAccessLogsTests(t framework.TestContext, expectLogs bool) {
+func TestAccessLogsDefaultProvider(t *testing.T) {
+	framework.NewTest(t).
+		Features("observability.telemetry.logging.defaultprovider").
+		Run(func(t framework.TestContext) {
+			t.NewSubTest("disabled").Run(func(t framework.TestContext) {
+				cfg := `
+accessLogFile: "/dev/null"
+`
+				ist := *(common.GetIstioInstance())
+				istio.PatchMeshConfigOrFail(t, ist.Settings().SystemNamespace, t.Clusters(), cfg)
+				runAccessLogsTests(t, false)
+				cfg = `
+accessLogFile: "/dev/stdout"
+`
+				istio.PatchMeshConfigOrFail(t, ist.Settings().SystemNamespace, t.Clusters(), cfg)
+			})
+			t.NewSubTest("enabled").Run(func(t framework.TestContext) {
+				cfg := `
+accessLogFile: "/dev/null"
+defaultProviders:
+  accessLogging:
+  - envoy
+`
+				ist := *(common.GetIstioInstance())
+				istio.PatchMeshConfigOrFail(t, ist.Settings().SystemNamespace, t.Clusters(), cfg)
+				runAccessLogsTests(t, true)
+				cfg = `
+accessLogFile: "/dev/stdout"
+defaultProviders:
+  accessLogging: []
+`
+				istio.PatchMeshConfigOrFail(t, ist.Settings().SystemNamespace, t.Clusters(), cfg)
```

Comment:
what is this doing? If its trying to revert the change,PatchMeshConfigOrFail automatically does that?

Reply:
you're right, the code is move from a really branch, in prev context there's no `PatchMeshConfigOrFail`, we do not need it now.

Reply:
fixed